content/200-concepts/100-components/02-prisma-client/058-transactions.mdx

```diff
@@ -13,6 +13,17 @@ A database transaction refers to a sequence of read/write operations that are _g
 
 </TopBlock>
 
+## About transactions in Prisma
+
+Prisma provides the following options for using transactions:
+
+- **Nested writes**: use the Prisma Client API to process multiple operations on one or more related records inside the same transaction.
+- **Batch/bulk transactions**: to process one or more operations in bulk at a time, use the bulk operations `updateMany`, `deleteMany`, and `createMany`.
+- **The `$transactions` API call**: use this call to either:
+  - pass an array of Prisma Client queries to be executed sequentially inside a transaction
+    OR
+  - pass a function that can contain arbitrary code including Prisma Client queries, non-Prisma code and other control flow to be executed in a transaction.
```

Comment:
```suggestion
  - `$transaction<R>(fn: (prisma: PrismaClient) => R): R`: Pass a function that can contain arbitrary code including Prisma Client queries, non-Prisma code and other control flow to be executed in a transaction.
```

Reply:
@millsp I think the word "arbitrary" here is disconcerting... the idea of arbitrary code sounds a bit worrying. ;-) I get it that ere it just means "any code" but I'd like ot remove it, unless there is a very specific need for it?

Reply:
Hmm, nice catch! I'd say that in this context it means "user code".

Reply:
okeeee,,,, I'll just leave it as "code."